OpenEdX/DI/ScreenAssembly.swift

```diff
@@ -270,7 +270,17 @@ class ScreenAssembly: Assembly {
                 courseEnd: courseEnd,
                 enrollmentStart: enrollmentStart,
                 enrollmentEnd: enrollmentEnd,
-                coreAnalytics: r.resolve(CoreAnalytics.self)!
+                coreAnalytics: r.resolve(CoreAnalytics.self)!,
+                courseDatesViewModel: CourseDatesViewModel(
+                    interactor: r.resolve(CourseInteractorProtocol.self)!,
+                    router: r.resolve(CourseRouter.self)!,
+                    cssInjector: r.resolve(CSSInjector.self)!,
+                    connectivity: r.resolve(ConnectivityProtocol.self)!,
+                    config: r.resolve(ConfigProtocol.self)!,
+                    courseID: courseID,
+                    courseName: courseName,
+                    analytics: r.resolve(CourseAnalytics.self)!
+                )
```

Comment:
Why do we create `CourseDatesViewModel` manually instead of injecting it via `r.resole(...)`?
And with this approach, we will have 2 different instances of CourseDatesViewModel for CourseHome. Is this ok?

Reply:
<img width="520" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/8323f1af-3fc7-4afc-8341-83c273547c05">

Maybe we can pass here the previously injected VM?

<img width="749" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/8562db49-9bdf-4da8-9267-b6a892400d50">